### PR TITLE
dsc gets a shutdown endpoint

### PR DESCRIPTION
**Pull request by leftwo**
Added a new endpoint to dsc to allow remote shutdown of all downstairs and then the dsc process itself.  This opens the door to more bash script replacement.

Updated the communication channel between the dsc main task and the sub-tasks for each downstairs to allow more than one outstanding command at a time.

Updated the test_dsc.sh to test (and make use of) the /shutdown endpoint.

### CODE REVIEW

File: dsc/src/control.rs [hunk 1]
```diff
@@ -356,6 +357,23 @@ async fn dsc_enable_restart_all(
     Ok(HttpResponseUpdatedNoContent())
 }
 
+/**
+ * Stop all downstairs, then stop ourselves.
+ */
+#[endpoint {
+    method = GET,
```
Comment (jmpesp) [inline dsc/src/control.rs:364]: should be POST

File: dsc/src/main.rs [hunk 2]
```diff
@@ -594,7 +619,7 @@ async fn start_dsc(
         println!("start ds: {:?}", ds.port);
         let txc = tx.clone();
         let dsc = ds.clone();
-        let (action_tx, action_rx) = watch::channel(Action::Start);
```
Comment (jmpesp) [inline dsc/src/main.rs:597]: `Action::Start` initialized the channel before, does it now need to be sent manually? should it be sent on the next line?

**Review by jmpesp — APPROVED** [review]
:package:

File: dsc/src/control.rs (hunk 1 above)
Comment (leftwo) [reply dsc/src/control.rs:364]: done

File: dsc/src/main.rs (hunk 2 above)
Comment (leftwo) [reply dsc/src/main.rs:597]: The downstairs start happens automatically in the `ds_start_monitor` function.  That the `Action::Start` was being pre-loaded in the channel before was not an actual requirement for starting, and was either a requirement of the `watch::channel` or a bug in my understanding of how to setup the channel.

**Pull request merged**